### ISSUE TITLE
How to use with another container extension

### ISSUE THREAD

**Issue by sebast889** [stoically/temporary-containers#261]
I'm trying to use another container extension https://addons.mozilla.org/en-US/firefox/addon/block-sites-outside-container/ together with this one but when setting a website in that extension to open in multiple permanent containers every time temporary ones take over so it never opens in the permanent ones. Is there any way to make them work together?

**Comment by stoically** [comment]
`Block sites outside container` would need to provide an API so other extensions can check whether a specific url is assigned, which would allow TC to ignore those urls - similar to https://github.com/mozilla/multi-account-containers/pull/1095 and https://github.com/kintesh/containerise/pull/18. To see if the maintainer is open to such an API, opening an issue over in his repo asking about it would be helpful.

**Comment by stoically** [comment]
Also, as a workaround, if you're using Isolation and that causes the conflict, you could add the urls to the `Per Domain Isolation` and set them to `Never` accordingly.

**Comment by sebast889** [comment]
I will try ask the maintainer and post back if I get a response.

**Comment by PinkPandaKatie** [comment]
Hi, I'm the author of the "Block Sites Outside Container" extension. I can add this API, but I need to know what information it needs to return. Do you just need to know whether a rule exists for a given domain, or do you need to know which containers it is set to open in?

**Comment by PinkPandaKatie** [comment]
I just uploaded a new version with the API. Here is how you can use it:

```
async function rule_exists(domain) {
    try {
        let response = await browser.runtime.sendMessage('block_outside_container@jspenguin.org', {action: "rule_exists", domain: domain});
        return response.rule_exists;
    } catch (err) {
        return false;
    }
}
```

**Comment by stoically** [comment]
That was fast, thanks! Published version 0.98 with the compatibility fix.

**Comment by sebast889** [comment]
@PinkPandaKatie @stoically

Thank you both for the fix and for the great extensions!
